extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -7,28 +7,121 @@
  * @package Jetpack
  */
 
-if (
-	( defined( 'IS_WPCOM' ) && IS_WPCOM ) ||
-	class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
-) {
-	jetpack_register_block(
-		'jetpack/tiled-gallery',
-		array(
-			'render_callback' => 'jetpack_tiled_gallery_load_block_assets',
-		)
-	);
+/**
+ * Jetpack Tiled Gallery Block class
+ *
+ * @since 7.4
+ */
+class Jetpack_Tiled_Gallery_Block {
+	/* Values for building srcsets */
+	const IMG_SRCSET_WIDTH_MAX  = 2000;
+	const IMG_SRCSET_WIDTH_MIN  = 600;
+	const IMG_SRCSET_WIDTH_STEP = 300;
 
 	/**
-	 * Tiled gallery block registration/dependency declaration.
+	 * Register the block
+	 */
+	public static function register() {
+		jetpack_register_block(
+			'jetpack/tiled-gallery',
+			array(
+				'render_callback' => array( __CLASS__, 'render' ),
+			)
+		);
+	}
+
+	/**
+	 * Tiled gallery block registration
 	 *
 	 * @param array  $attr    Array containing the block attributes.
 	 * @param string $content String containing the block content.
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	public static function render( $attr, $content ) {
 		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery' );
 
+		$is_squareish_layout = self::is_squareish_layout( $attr );
+
+		if ( preg_match_all( '/<img [^>]+>/', $content, $images ) ) {
+			/**
+			 * This block processes all of the images that are found and builds $find and $replace.
+			 *
+			 * The original img is added to the $find array and the replacement is made and added
+			 * to the $replace array. This is so that the same find and replace operations can be
+			 * made on the entire $content.
+			 */
+			$find    = array();
+			$replace = array();
+
+			foreach ( $images[0] as $image_html ) {
+				if (
+					preg_match( '/data-width="([0-9]+)"/', $image_html, $img_height )
+					&& preg_match( '/data-height="([0-9]+)"/', $image_html, $img_width )
+					&& preg_match( '/src="([^"]+)"/', $image_html, $img_src )
+				) {
+					// Drop img src query string so it can be used as a base to add photon params
+					// for the srcset.
+					$src_parts   = explode( '?', $img_src[1], 2 );
+					$orig_src    = $src_parts[0];
+					$orig_height = absint( $img_height[1] );
+					$orig_width  = absint( $img_width[1] );
+
+					if ( ! $orig_width || ! $orig_height || ! $orig_src ) {
+						continue;
+					}
+
+					$srcset_parts = array();
+					if ( $is_squareish_layout ) {
```

Comment:
Wouldn't we at this point just see square images anyway since that's how they're saved from the editor, without even checking for the type of layout?

I'm hoping we would not need this `if/else` here at all and this section would be more simple.

This might be useful here, too: https://developer.wordpress.org/reference/functions/wp_constrain_dimensions/

Reply:
Removing the branch with a lot of very similar code is appealing, but also a fair bit of work for little benefit. I looked at it when implementing these changes and considered this was good enough.

I can take another look (or you're welcome to) if you think it's worth it.

Reply:
Nah, merging and deploying is more fun than reworking this out. 👍